build-system/tasks/report-test-results.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file implements the `gulp report-test-results` task, which [[does something]]
+ */
+
+'use strict';
+
+const fetch = require('node-fetch');
+const fs = require('fs').promises;
+const log = require('fancy-log');
+const path = require('path');
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../common/travis');
+const {cyan, green, yellow} = require('ansi-colors');
+
+const REPORTING_API_URL = 'https://amp-test-cases.appspot.com/report';
+
+async function getReport(testType) {
+  const karmaReportString = await fs.readFile(
+    `result-reports/${testType}.json`
```

Comment:
Hm, on second thought, I think a helper for the target directory name would be better. We _will_ have to import the function in runtime-test-base.js; however, modularity is not harmed by doing that import -- modularity was already harmed by the fact that the file names need to agree in both places.

Put another way, having a raw `result-reports/${config.testType}.json` in runtime-test-base.js isn't more or less modular than making a helper function and importing it there, because the end result is about the same -- both files have to agree on the filename, so they're (a little) coupled in that way.

It may be better to have the helper method in a separate file so that this file is just in charge of the gulp task itself; it may also be better to just not worry about it, since the filename is used in like 3 or 4 places.

Thoughts? @rcebulko @estherkim

Reply:
Unfortunately some build-system code just relies on the location of some sets of files. At most I could see exporting a constant `RESULT_REPORT_DIRECTORY` to be used elsewhere, but I'll defer entirely to Esther if she disagrees as I have less familiarity with the conventions in build-system.

Reply:
Let's leave it as is for now. Generally I like in-place strings more than exported consts because it's way easier to read.

But if it makes sense to have it all in 1 place, we can optimize later.